docs: add docs on how to configure models deployed via azure (#337)
simonw/llm: docs/other-models.md

llm logs -n 1
```

### Azure OpenAI Service

> LLM supports OpenAI models deployed via [Azure OpenAI Service](https://azure.microsoft.com/en-us/products/ai-services/openai-service). These can also be added using the same configuration mechanism.

Reviewer: ```suggestion
LLM supports OpenAI models deployed via [Azure OpenAI Service](https://azure.microsoft.com/products/ai-services/openai-service). These can also be added using the same configuration mechanism.
```

Always better to use the global link and let it redirect as needed.

> The `model_id` is the name LLM will use for the model. The `model_name` is the deployment name of the model on Azure OpenAI Service. 

Reviewer: ```suggestion
The `model_id` is the name LLM will use for the model. The `model_name` is the name of the model on Azure OpenAI Service, like `gpt-35-turbo-16k`.
```

The `api_base` will be the Azure OpenAI Service endpoint. The `api_key_name` is the name of the Azure OpenAI Service API key stored using the {ref}`api-keys` feature.

You must add `api_version` to use a specific version of the Azure OpenAI Service API.

> Set `api_type` to `azure` to specify that the OpenAI client library should use the Azure OpenAI Service API.

Reviewer: Worth noting that "azure_ad" is not supported, as far as I can tell (keyless auth with AAD).

Reviewer: ```suggestion
Set `api_type` to `azure` to specify that the OpenAI client library should use the Azure OpenAI Service API. Note that the type of `azure_ad` (keyless auth) is not yet supported.
```

> You must also add `api_engine` to specify the base model to use, eg: `gpt-35-turbo-16k`

Reviewer: ```suggestion
You must also add `api_engine` to specify the name of the deployment.
```

For example, to add the `mygpt35` model (a `gpt-35-turbo-16k` model) deployed on Azure OpenAI Service, using the `2023-05-15` version of the API, with the account endpoint `https://my-openai-account.openai.azure.com/`, add this to your `extra-openai-models.yaml` file:

```yaml
- model_id: mygpt

>   model_name: mygpt35

Reviewer: ```suggestion
  model_name: gpt-35-turbo
```

This seems to have model and deployment reversed.

  api_base: https://my-openai-account.openai.azure.com/
  api_key_name: azure
  api_version: 2023-05-15
  api_type: azure

>   api_engine: gpt-35-turbo-16k

Reviewer: ```suggestion
  api_engine: mygptchat
```
